packages/runtime/src/useCases/consumption/attributes/DeleteRepositoryAttribute.ts

```diff
@@ -0,0 +1,48 @@
+import { Result } from "@js-soft/ts-utils";
+import { AttributesController, LocalAttribute } from "@nmshd/consumption";
+import { AccountController, CoreId } from "@nmshd/transport";
+import { Inject } from "typescript-ioc";
+import { AttributeIdString, RuntimeErrors, SchemaRepository, SchemaValidator, UseCase } from "../../common";
+
+export interface DeleteRepositoryAttributeRequest {
+    attributeId: AttributeIdString;
+}
+
+class Validator extends SchemaValidator<DeleteRepositoryAttributeRequest> {
+    public constructor(@Inject schemaRepository: SchemaRepository) {
+        super(schemaRepository.getSchema("DeleteRepositoryAttributeRequest"));
+    }
+}
+
+export class DeleteRepositoryAttributeUseCase extends UseCase<DeleteRepositoryAttributeRequest, void> {
+    public constructor(
+        @Inject private readonly attributesController: AttributesController,
+        @Inject private readonly accountController: AccountController,
+        @Inject validator: Validator
+    ) {
+        super(validator);
+    }
+
+    protected async executeInternal(request: DeleteRepositoryAttributeRequest): Promise<Result<void>> {
+        const repositoryAttribute = await this.attributesController.getLocalAttribute(CoreId.from(request.attributeId));
+
+        if (typeof repositoryAttribute === "undefined") {
+            return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
+        }
+
+        if (!repositoryAttribute.isRepositoryAttribute(this.accountController.identity.address)) {
+            return Result.fail(RuntimeErrors.attributes.isNotRepositoryAttribute(CoreId.from(request.attributeId)));
+        }
+
+        const validationResult = await this.attributesController.validateSourceAttributeDeletion(repositoryAttribute);
+        if (validationResult.isError()) {
+            return Result.fail(validationResult.error);
+        }
+
+        await this.attributesController.deleteRepositoryAttribute(repositoryAttribute);
```

Comment:
as the actual method is now called `deleteRepositoryAttribute` why do we need all of the validations before? Shouldn't they be triggered inside the method?

Reply:
Yes, the validations are performed inside `deleteRepositoryAttribute`. I thought it would be nice to wrap the consumptions errors, according to https://github.com/nmshd/runtime/blob/main/packages/runtime/src/useCases/consumption/attributes/SucceedRepositoryAttribute.ts#L55 .

Reply:
IC, better safe than sorry :)